docs/index.rst

```diff
@@ -115,6 +115,7 @@ anonymous sources.
    development/contributor_guidelines
    development/tips_and_tricks
    development/database_migrations
+   development/translations
    development/i18n
    development/l10n
```

Comment:
Just adding a reminder here to remove the old pre-reorg doc :)

Reply:
I removed `development/l10n.rst` which is now redundant. (On "translations" vs "localization", I think the document is specifically about translations, but I don't feel strongly about it.)